### PR TITLE
plugins/treesitter: rename our injection file to nixvim-injections.scm

### PR DESCRIPTION
**Pull request by GaetanLepage**
This removes the following issue when enabling `performance.combinePlugins.standalonePlugins`:
```
> error: collision between `/nix/store/71m5cmbwqbdhyriphml046qq7299s8d7-vimplugin-hmts.nvim-2024-05-07/queries/nix/injections.scm' and `/nix/store/4l7v3q9wlkqjfmrzy6m9k5qpb77b4c74-vimplugin-treesitter-grammar-nix/queries/nix/injections.scm'
```

### CODE REVIEW

File: plugins/languages/treesitter/nixvim-injections.scm
Comment (MattSturgeon) [inline plugins/languages/treesitter/nixvim-injections.scm:1]: We don't need to rename the file in our repo to affect the name in the /nix/store.

Unless you just think this is a better name?

File: tests/test-sources/plugins/languages/treesitter/combine-plugins.nix [hunk 1]
```diff
@@ -3,12 +3,7 @@
   combine-plugins = {
     performance.combinePlugins.enable = true;
 
-    plugins.treesitter = {
-      enable = true;
-
-      # Exclude nixvim injections for test to pass
-      nixvimInjections = false;
-    };
+    plugins.treesitter.enable = true;
```
Comment (MattSturgeon) [inline tests/test-sources/plugins/languages/treesitter/combine-plugins.nix:6]: Why change the test?

File: plugins/languages/treesitter/nixvim-injections.scm
Comment (GaetanLepage) [reply plugins/languages/treesitter/nixvim-injections.scm:1]: That's true indeed. Even though, I find this name to be clearer. If it's OK for you we can merge this as is.

**Comment by GaetanLepage** [comment]
I still get a clash between the `injections.scm` file from the `hmts` plugin and the one from the `nix` treesitter grammar.

**Comment by GaetanLepage** [comment]
And I fear that this PR breaks the injection...

**Comment by khaneliman** [comment]
Can we just use `vim.treesitter.query.set("nix", "injections", "(comment) @comment")`

**Comment by stasjok** [comment]
I don't get it. Why do you change nixvim injections, if based on your error you have entirely different problem. You can't have collisions with filesPlugin (what you are trying to change here), because I excluded it. You have collisions with this plugin: https://github.com/calops/hmts.nvim. So you can add "hmts.nvim" to standalonePlugins.

**Comment by GaetanLepage** [comment]
> I don't get it. Why do you change nixvim injections, if based on your error you have entirely different problem. You can't have collisions with files added via `extraFiles` (what you are trying to change here), because I excluded it. You have collisions with this plugin: https://github.com/calops/hmts.nvim. So you can add "hmts.nvim" to standalonePlugins.

You are right, this doesn't address my issue at all.
Closing.

**Pull request closed without merging**